cache.go

```diff
@@ -241,31 +243,24 @@ func (c *Cache[K, V]) GetOrSet(key K, val V, opts ...ItemOption) (actual V, load
 
 // DeleteExpired all expired items from the cache.
 func (c *Cache[K, V]) DeleteExpired() {
-	c.mu.Lock()
-	l := c.expManager.len()
-	c.mu.Unlock()
-
 	evict := func() bool {
-		key := c.expManager.pop()
-		// if is expired, delete it and return nil instead
-		item, ok := c.cache.Get(key)
-		if ok {
-			if item.Expired() {
-				c.cache.Delete(key)
-				return false
-			}
-			c.expManager.update(key, item.Expiration)
+		c.mu.Lock()
+		defer c.mu.Unlock()
+		if c.expManager.len() == 0 {
+			return false
+		}
+		key, expiration := c.expManager.pop()
+		if nowFunc().After(expiration) {
```

Comment:
I think better to use expired than this because a Expired method already has checking expiration logic.

Reply:
This code was meant to avoid an extra cache.Get(). Also, if the cache already had deleted the item, Expired() was not callable. However, your approach is more maintainable, since if you edit the way Expired() works you need to edit this code as well. So what should I do about this change ?